arangod/Aql/OptimizerRuleReplaceEqualAttributeAccess.cpp

```diff
@@ -0,0 +1,308 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2023 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+////////////////////////////////////////////////////////////////////////////////
+
+#include "ApplicationFeatures/ApplicationServer.h"
+#include "Aql/AstHelper.h"
+#include "Aql/AttributeNamePath.h"
+#include "Aql/Collection.h"
+#include "Aql/Condition.h"
+#include "Aql/ConditionFinder.h"
+#include "Aql/DocumentProducingNode.h"
+#include "Aql/ExecutionEngine.h"
+#include "Aql/ExecutionNode.h"
+#include "Aql/ExecutionPlan.h"
+#include "Aql/Expression.h"
+#include "Aql/IndexNode.h"
+#include "Aql/IndexStreamIterator.h"
+#include "Aql/JoinNode.h"
+#include "Aql/Optimizer.h"
+#include "Aql/OptimizerUtils.h"
+#include "Aql/Projections.h"
+#include "Aql/Query.h"
+#include "Aql/Variable.h"
+#include "Aql/types.h"
+#include "Basics/AttributeNameParser.h"
+#include "Basics/NumberUtils.h"
+#include "Basics/ScopeGuard.h"
+#include "Basics/StaticStrings.h"
+#include "Containers/FlatHashSet.h"
+#include "Containers/NodeHashMap.h"
+#include "Containers/HashSet.h"
+#include "Containers/SmallUnorderedMap.h"
+#include "Containers/SmallVector.h"
+#include "Geo/GeoParams.h"
+#include "Graph/ShortestPathOptions.h"
+#include "Graph/TraverserOptions.h"
```

Comment:
nit: many of these includes seem to be unnecessary in this file, so they could be cleaned up.